src/coreclr/src/inc/corinfoinstructionset.h

```diff
@@ -0,0 +1,276 @@
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// DO NOT EDIT THIS FILE! IT IS AUTOGENERATED
+// FROM /src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt
+// using /src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/gen.bat
+
+#ifndef CORINFOINSTRUCTIONSET_H
+#define CORINFOINSTRUCTIONSET_H
+
+enum CORINFO_InstructionSet
+{
+    InstructionSet_ILLEGAL = 0,
+    InstructionSet_NONE = 63,
+#ifdef TARGET_ARM64
+    InstructionSet_ArmBase=1,
+    InstructionSet_ArmBase_Arm64=2,
+    InstructionSet_AdvSimd=3,
+    InstructionSet_AdvSimd_Arm64=4,
+    InstructionSet_Aes=5,
+    InstructionSet_Crc32=6,
+    InstructionSet_Crc32_Arm64=7,
+    InstructionSet_Sha1=8,
+    InstructionSet_Sha256=9,
+    InstructionSet_Atomics=10,
+    InstructionSet_Vector64=11,
+    InstructionSet_Vector128=12,
+#endif // TARGET_ARM64
+#ifdef TARGET_AMD64
+    InstructionSet_SSE=1,
+    InstructionSet_SSE2=2,
+    InstructionSet_SSE3=3,
+    InstructionSet_SSSE3=4,
+    InstructionSet_SSE41=5,
+    InstructionSet_SSE42=6,
+    InstructionSet_AVX=7,
+    InstructionSet_AVX2=8,
+    InstructionSet_AES=9,
+    InstructionSet_BMI1=10,
+    InstructionSet_BMI2=11,
+    InstructionSet_FMA=12,
+    InstructionSet_LZCNT=13,
+    InstructionSet_PCLMULQDQ=14,
+    InstructionSet_POPCNT=15,
+    InstructionSet_Vector128=16,
+    InstructionSet_Vector256=17,
+    InstructionSet_BMI1_X64=18,
+    InstructionSet_BMI2_X64=19,
+    InstructionSet_LZCNT_X64=20,
+    InstructionSet_POPCNT_X64=21,
+    InstructionSet_SSE_X64=22,
+    InstructionSet_SSE2_X64=23,
+    InstructionSet_SSE41_X64=24,
+    InstructionSet_SSE42_X64=25,
+#endif // TARGET_AMD64
+#ifdef TARGET_X86
+    InstructionSet_SSE=1,
+    InstructionSet_SSE2=2,
+    InstructionSet_SSE3=3,
+    InstructionSet_SSSE3=4,
+    InstructionSet_SSE41=5,
+    InstructionSet_SSE42=6,
+    InstructionSet_AVX=7,
+    InstructionSet_AVX2=8,
+    InstructionSet_AES=9,
+    InstructionSet_BMI1=10,
+    InstructionSet_BMI2=11,
+    InstructionSet_FMA=12,
+    InstructionSet_LZCNT=13,
+    InstructionSet_PCLMULQDQ=14,
+    InstructionSet_POPCNT=15,
+    InstructionSet_Vector128=16,
+    InstructionSet_Vector256=17,
+    InstructionSet_BMI1_X64=18,
```

Comment:
Just making sure, do these need to be defined so x86 can report `false`?

Reply:
Yeah. I got everything working, but I still couldn't build x86 …. and the amount of idefs that were going to be required was entirely unreasonable.